simulator/examples/text-transparent.rs

```diff
@@ -3,58 +3,54 @@
 //! Demonstrate the background styles and transparency behaviors of different font styles.
 
 use embedded_graphics::{
-    egcircle, egrectangle, egtext, fonts::Font6x8, pixelcolor::Rgb565, prelude::*, primitive_style,
-    text_style,
+    fonts::{Font6x8, Text},
+    pixelcolor::Rgb565,
+    prelude::*,
+    primitives::{Circle, Rectangle},
+    style::{PrimitiveStyleBuilder, TextStyleBuilder},
 };
 use embedded_graphics_simulator::{OutputSettingsBuilder, SimulatorDisplay, Window};
 
 fn main() -> Result<(), core::convert::Infallible> {
     let mut display: SimulatorDisplay<Rgb565> = SimulatorDisplay::new(Size::new(256, 128));
 
-    egcircle!(
-        top_left = (0, 0),
-        diameter = 41,
-        style = primitive_style!(fill_color = Rgb565::RED)
-    )
-    .into_iter()
-    .chain(&egrectangle!(
-        top_left = (20, 20),
-        size = (80, 60),
-        style = primitive_style!(fill_color = Rgb565::RED)
-    ))
-    .draw(&mut display)
-    .unwrap();
-
-    egtext!(
-        text = "Hello world! - no background",
-        top_left = (15, 15),
-        style = text_style!(font = Font6x8, text_color = Rgb565::WHITE,)
-    )
-    .draw(&mut display)
-    .unwrap();
-
-    egtext!(
-        text = "Hello world! - filled background",
-        top_left = (15, 30),
-        style = text_style!(
-            font = Font6x8,
-            text_color = Rgb565::YELLOW,
-            background_color = Rgb565::BLUE
+    Circle::new(Point::new(0, 0), 41)
+        .into_styled(PrimitiveStyleBuilder::new().fill_color(Rgb565::RED).build())
+        .into_iter()
+        .chain(
+            &Rectangle::new(Point::new(20, 20), Size::new(80, 60))
+                .into_styled(PrimitiveStyleBuilder::new().fill_color(Rgb565::RED).build()),
```

Comment:
```suggestion
                .into_styled(PrimitiveStyle::with_fill(Rgb565::RED)),
```
Or share the style with the previous call by creating a variable.